pkg/spec/types.go

```diff
@@ -28,6 +28,16 @@ const (
 	EventSync   EventType = "SYNC"
 )
 
+type RoleOrigin int
+
+const (
+	RoleOriginSystem = iota
+	RoleOriginInfrastructure
+	RoleOriginManifest
+	RoleOriginTeamsAPI
+	RoleOriginUnknown
+)
+
```

Comment:
we have this defined in  two separate PRs. Shall we merge [that one](https://github.com/zalando-incubator/postgres-operator/pull/206) first ?